File: kbmf1k1k/kbmf1k1k_supervised_regression_variational_test.m
function prediction = kbmf1k1k_supervised_regression_variational_test(Kx, Kz, state)
    prediction.Gx.mu = state.Ax.mu' * Kx;

    prediction.Gz.mu = state.Az.mu' * Kz;

    prediction.Y.mu = prediction.Gx.mu' * prediction.Gz.mu;
end

File: kbmf1k1k/kbmf1k1k_supervised_regression_variational_test.R
# Mehmet Gonen (mehmet.gonen@gmail.com)

kbmf1k1k_supervised_regression_variational_test <- function(Kx, Kz, state) {
  Gx <- list(mu = crossprod(state$Ax$mu, Kx))

  Gz <- list(mu = crossprod(state$Az$mu, Kz))

  Y <- list(mu = crossprod(Gx$mu, Gz$mu))

  prediction <- list(Gx = Gx, Gz = Gz, Y = Y)
}